src/renderer/components/dialogs/DialogController.tsx

```diff
@@ -66,27 +65,36 @@ export type DialogProps = {
 
 var dialogCounter = 1
 
-export class Controller extends React.Component<
+export type OpenDialogFunctionType = typeof DialogController.prototype.openDialog
+export type CloseDialogFunctionType = typeof DialogController.prototype.closeDialog
+
+export default class DialogController extends React.Component<
   any,
   {
-    dialogs: dialogs
+    dialogs: {
+      [key: string]: Dialog
+    }
   }
 > {
   constructor(props: any) {
     super(props)
 
     this.state = { dialogs: {} }
-    this.close = this.close.bind(this)
+    this.openDialog = this.openDialog.bind(this)
+    this.closeDialog = this.closeDialog.bind(this)
   }
 
-  open(fnc: any, props?: todo /* infer from component */) {
+  openDialog(
+    fnc: DialogId | ((props: DialogProps) => React.ReactElement),
```

Comment:
```suggestion
    fnc: DialogId | ((props: DialogProps & DialogAdditionProps) => React.ReactElement),
```
probably

Reply:
I think the infer thing makes more sense to do when we converted all dialogs to functional components.